### PR TITLE
Modify list command

### LINKED ISSUE #167: [PE Dry Run]Listed modules does not show
<hr>

**Reported by:** @benedictcss
**Severity:** Not Specified

<sub>[original: nus-cs2103-AY1819S1/pe-dry-run#266]</sub>

### LINKED ISSUE #179: [PE Dry Run]List Command does nothing
As the user guide has no diagram to show what 'list command' execution should look like. I am not entirely sure what the command does, however right now, there is no result displayed.

If it should matter, the current user setup is as follows
`setup y/2 s/1 m/Computer Science f/Software Engineering`

**Replicating the bug**
The picture below shows the module I have in y1s1

list_show_havey1s1

I then switch to the Y1S2 window and executes the command `list`

Before

list_before_cmd

After (nothing happens)

list_after_cmd

<hr>

**Reported by:** @gilgameshtc
**Severity:** `High`

<sub>[original: nus-cs2103-AY1819S1/pe-dry-run#1044]</sub>

### PR DESCRIPTION
**Pull request by Hilda-Ang**
Fixes #167 
Fixes #179

### CODE REVIEW

File: src/main/java/seedu/planner/model/ModelManager.java [hunk 1]
```diff
@@ -48,12 +49,12 @@ public ModelManager(ReadOnlyModulePlanner modulePlanner, UserPrefs userPrefs) {
 
         versionedModulePlanner = new VersionedModulePlanner(modulePlanner);
 
-        takenModules = new ArrayList<>();
+        takenModulesPerSemester = new ArrayList<>();
         for (int i = 0; i < MAX_NUMBER_SEMESTERS; i++) {
-            takenModules.add(new SortedList<>(versionedModulePlanner.getTakenModules(i), (x, y) ->
-                    x.compareTo(y)));
+            takenModulesPerSemester.add(new SortedList<>(versionedModulePlanner.getTakenModules(i)));
```
Comment (GabrielYik) [inline src/main/java/seedu/planner/model/ModelManager.java]: Are the taken modules still sorted? Cos I removed the `x.compareTo(y)` and it wasn't sorted leh

File: src/main/java/seedu/planner/model/util/IndexUtil.java [hunk 2]
```diff
@@ -75,4 +77,24 @@ public static String convertIndexToYearAndSemester(int index) {
 
         return String.valueOf(year) + String.valueOf(semester);
     }
+
+    public static int[] getIndicesFromYear(int year) {
+        int[] indices = new int[2];
+
+        if (year == 1) {
+            indices[0] = 0;
```
Comment (GabrielYik) [inline src/main/java/seedu/planner/model/util/IndexUtil.java]: Might be good to give a name to `0` and `1`

File: src/main/java/seedu/planner/ui/MainWindow.java [hunk 3]
```diff
@@ -307,4 +313,10 @@ private void handleSuggestModule(SuggestModuleEvent event) {
                 event.getIndex(), ModulePanelType.SUGGESTED);
         suggestedModulesPlaceholder.getChildren().set(CURRENT_NODE, panel.getRoot());
     }
+
+    @Subscribe
+    private void handleListEvent(ListModuleEvent event) {
+        ModuleListPanel panel = takenModuleListPanels.get(MAX_NUMBER_SEMESTERS);
+        takenModulesPlaceholder.getChildren().set(CURRENT_NODE, panel.getRoot());
```
Comment (GabrielYik) [inline src/main/java/seedu/planner/ui/MainWindow.java]: The use of `CURRENT_NODE` is not really "safe" cos not panels have their child at index `1`. My mistake. Best to use `MainWindow#setPlaceHolder` in PR I merged

File: src/test/java/seedu/planner/logic/commands/ListCommandTest.java [hunk 4]
```diff
@@ -67,7 +64,7 @@ public void equals() {
         // null -> returns false
         assertFalse(listFirstCommand.equals(null));
 
-        // different index -> returns false
+        // different  -> year returns false
```
Comment (GabrielYik) [inline src/test/java/seedu/planner/logic/commands/ListCommandTest.java]: `year ->`?

File: src/main/java/seedu/planner/model/ModelManager.java (hunk 1 above)
Comment (Hilda-Ang) [reply src/main/java/seedu/planner/model/ModelManager.java]: Oh ya just realised, sorry I'll put it back haha

File: src/main/java/seedu/planner/model/util/IndexUtil.java [hunk 5]
```diff
@@ -75,4 +77,27 @@ public static String convertIndexToYearAndSemester(int index) {
 
         return String.valueOf(year) + String.valueOf(semester);
     }
+
+    public static int[] getIndicesFromYear(int year) {
+        int[] indices = new int[2];
+
+        final int indexZero = 0;
```
Comment (GabrielYik) [inline src/main/java/seedu/planner/model/util/IndexUtil.java]: I actually meant something like `FIRST_SEMESTER_INDEX` and `SECOND_SEMESTER_INDEX` haha. Cos I didn't really understand what 0 and 1 were at first glance

**Pull request merged**